src/main/java/org/graylog2/gelfclient/transport/GelfSenderThread.java

```diff
@@ -139,4 +146,41 @@ public void stop() {
         keepRunning.set(false);
         senderThread.interrupt();
     }
+
+    /**
+     * Attempt to send all messages in the queue before allowing shutdown to proceed.
+     * Retrying will continue for the indicated {@code SHUTDOWN_SENDING_RETRIES} and {@code SHUTDOWN_SENDING_DELAY_MS.}
+     */
+    void flushAndStop() {
```

Comment:
@bernd This is my first pass at adding a stop method that flushes messages to the network before shutdown.